lib/suppressions.cpp

```diff
@@ -334,15 +334,22 @@ bool SuppressionList::Suppression::parseComment(std::string comment, std::string
     if (comment.size() < 2)
         return false;
 
-    if (comment.find(';') != std::string::npos)
-        comment.erase(comment.find(';'));
-
-    if (comment.find("//", 2) != std::string::npos)
-        comment.erase(comment.find("//",2));
-
     if (comment.compare(comment.size() - 2, 2, "*/") == 0)
         comment.erase(comment.size() - 2, 2);
 
+    std::string::size_type extraPos = comment.find(';');
+    std::string::size_type extraDelimiterSize = 1;
+
+    if (extraPos == std::string::npos) {
+        extraPos = comment.find("//", 2);
+        extraDelimiterSize = 2;
+    }
+
+    if (extraPos != std::string::npos) {
+        extraComment = trim(comment.substr(extraPos + extraDelimiterSize));
```

Comment:
It feels like it could be a good idea to remove special characters from `extraComment`. Otherwise we would have to ensure that they are encoded properly in the XML but I don't feel that is required.